packages/cli/src/scaling/worker-server.ts

```diff
@@ -87,51 +85,36 @@ export class WorkerServer {
 	}
 
 	private async mountEndpoints() {
-		if (this.endpointsConfig.health) {
-			this.app.get('/healthz', async (req, res) => await this.healthcheck(req, res));
+		const { health, overwrites, metrics } = this.endpointsConfig;
+
+		if (health) {
+			// @TODO: This now mirrors main's healthcheck but alters preexisting behavior for /healthz. List as breaking change?
```

Comment:
For discussion

Reply:
This change would mean that any rolling deployments of the worker containers would not wait until the new workers have a DB/redis connection setup. This could lead to short windows where executions are queuing up, but there are no workers actually processing them.
Considering that (unlike main and webhook) workers aren't user facing, I'm fine with this change.
But, maybe @tomi has some opinions on this as well.

Reply:
Yeah it's probably fine like Adi said. However, we had some discussion about what the healthchecks could/should be in [PAY-1903](https://linear.app/n8n/issue/PAY-1903/investigate-n8n-crash-and-recovery-failure-after-upgrade-to-1562). Instead of making this breaking change now, maybe we should overhaul all the healthchecks in one go so we don't need to introduce multiple breaking changes for this.

Reply:
In the context of workers, why would this be a breaking change?
I'd assume that worker health-check is an internal implementation detail, and this change should (hopefully) not create issues for any user.

Reply:
It changes the behaviour of the healthcheck. Basically the old implementation is moved to a new endpoint and the old endpoint gets a new behaviour